### ISSUE TITLE
Touch scroll inside results on mobile causing incorrect navigation

### ISSUE THREAD

**Issue by greatwitenorth** [algolia/algoliasearch-magento#1014]
**Bug: What is the current behavior?**
When I touch to scroll on mobile (tested on android Pixel 2) in search results, I'm instantly navigated to the url of the item I initially touched.

**Bug: What is the expected behavior?**
When I touch to scroll on mobile inside of search results, I expect to scroll lower into the results.

**Bug: What is the proposed solution?**
The problem code seems to be in ```js/algoliasearch/autocomplete.js```:
```
// Hack to handle buggy onclick event on iOS
$(algoliaConfig.autocomplete.selector).each(function () {
	var data = $(this).data('aaAutocomplete');
	var dropdown = data.dropdown;
	var suggestionClass = '.' + dropdown.cssClasses.prefix + dropdown.cssClasses.suggestion;

	dropdown.$menu.on('click', suggestionClass, function(e) {
		e.preventDefault();
		e.stopPropagation();

		var url = $(this).find('a').attr('href');
		location.assign(url);
	});
});
```
When I remove this code I'm able to scroll. The only issue is that if I do click the link, the search results disappear while I navigate to the new page. Not a huge deal, but better than having the user not being able to scroll at all.

**What is the version of Magento and of Algolia extension you are using? Always use the latest version of the extension one before opening a bug issue.**
Algolia v1.13
Magento v1.9.3.1

**Comment by greatwitenorth** [comment]
Also downgrading to Algolia v1.12 seems to fix the issue.

**Comment by Cmac2712** [comment]
I've also noticed this bug. I think it's to do with the fix put in place for this issue: https://github.com/algolia/algoliasearch-magento/issues/902

**Comment by laconica-roma** [comment]
@greatwitenorth , here is the solution
```javascript
                // Hack to handle buggy onclick event on iOS
		$(algoliaConfig.autocomplete.selector).each(function () {
			var data = $(this).data('aaAutocomplete');
			var dropdown = data.dropdown;
			var suggestionClass = '.' + dropdown.cssClasses.prefix + dropdown.cssClasses.suggestion;
			var touchmoved;

			dropdown.$menu.on('touchend', suggestionClass, function (e) {
				if (touchmoved != true) {
					e.preventDefault();
					e.stopPropagation();

					var url = $(this).find('a').attr('href');
					location.assign(url);
				}
			}).on('touchmove', function (e) {
				touchmoved = true;
			}).on('touchstart', function (e) {
				touchmoved = false;
			});
		});
```

P.S.
I can't contribute to the extension from windows :/

**Comment by JanPetr** [comment]
Hello everybody,

thanks for raising the issue. It's been fixed in #1009.
@greatwitenorth Can you apply the fix to your code?

@laconica-roma Thanks for jumping in with the fix! Can you elaborate more on the "Windows" thing? Maybe open a new issue about it. Thanks! 🚀
